sound/soc/sof/pcm.c

```diff
@@ -356,9 +366,25 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 
 		/* fallthrough */
 	case SNDRV_PCM_TRIGGER_START:
+		if (spcm->stream[substream->stream].suspend_ignored) {
+			/*
+			 * This case will be triggered when INFO_RESUME is
+			 * not supported, no need to re-start streams that
+			 * remained enabled in D0ix.
+			 */
+			spcm->stream[substream->stream].suspend_ignored = false;
+			return 0;
+		}
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		/* Don't suspend the d0i3 compatible stream at S0ix. */
```

Comment:
The comment shouldn't say "don't suspend, the stream is going to be in the SUSPENDED state nevertheless. We just dont stop the pipeline"